module/exporter-file/src/Infrastructure/Processor/ProductProcessor.php

```diff
@@ -0,0 +1,121 @@
+use Ergonode\ExporterFile\Domain\Entity\FileExportChannel;
+
+/**
+ */
+class ProductProcessor
+{
+    /**
+     * @var AttributeQueryInterface
+     */
+    private AttributeQueryInterface $attributeQuery;
+
+    /**
+     * @var TranslationInheritanceCalculator
+     */
+    private TranslationInheritanceCalculator $calculator;
+
+    /**
+     * @var AttributeRepositoryInterface
+     */
+    private AttributeRepositoryInterface $attributeRepository;
+
+    /**
+     * @param AttributeQueryInterface          $attributeQuery
+     * @param TranslationInheritanceCalculator $calculator
+     * @param AttributeRepositoryInterface     $attributeRepository
+     */
+    public function __construct(
+        AttributeQueryInterface $attributeQuery,
+        TranslationInheritanceCalculator $calculator,
+        AttributeRepositoryInterface $attributeRepository
+    ) {
+        $this->attributeQuery = $attributeQuery;
+        $this->calculator = $calculator;
+        $this->attributeRepository = $attributeRepository;
+    }
+
+    /**
+     * @param FileExportChannel $channel
+     * @param AbstractProduct   $product
+     *
+     * @return ExportData
+     *
+     * @throws ExportException
+     */
+    public function process(FileExportChannel $channel, AbstractProduct $product): ExportData
+    {
+        try {
+            $data = new ExportData();
+
+            $attributes = $this->attributeQuery->getDictionary();
+            asort($attributes);
+
+            foreach ($channel->getLanguages() as $language) {
+                $data->set($language, $this->getLanguage($product, $language, $attributes));
+            }
+
+            return $data;
+        } catch (\Exception $exception) {
+            throw new ExportException(
+                sprintf('Can\'t process export for %s', $product->getSku()->getValue()),
+                $exception
+            );
+        }
+    }
+
+    /**
+     * @param AbstractProduct $product
+     * @param Language        $language
+     * @param array           $attributes
+     *
+     * @return LanguageData
+     */
+    private function getLanguage(AbstractProduct $product, Language $language, array $attributes): LanguageData
+    {
+        $result = new LanguageData();
+        $result->set('_id', $product->getId()->getValue());
+        $result->set('_sku', $product->getSku()->getValue());
+        $result->set('_type', $product->getType());
+        $result->set('_template', $product->getTemplateId()->getValue());
+        foreach ($attributes as $attributeId => $code) {
+            $code = new AttributeCode($code);
+            if ($product->hasAttribute($code)) {
+                $attribute = $this->attributeRepository->load(new AttributeId($attributeId));
+                Assert::isInstanceOf($attribute, AbstractAttribute::class);
+                $value = $product->getAttribute($code);
+                $calculatedValue = $this->calculator->calculate($attribute, $value, $language);
```

Comment:
What about multimedia and select field? This is where the id will be set ?

Reply:
yes